extensions/json-language-features/client/src/jsonClient.ts

```diff
@@ -163,25 +164,26 @@ export async function startClient(context: ExtensionContext, newLanguageClient:
 		window.showInformationMessage(l10n.t('JSON schema cache cleared.'));
 	}));
 
+	toDispose.push(workspace.onDidSaveTextDocument(async (document) => {
+		const sortOnSave = workspace.getConfiguration().get<boolean>(SettingIds.enableSortOnSave);
+		if (sortOnSave && (document.languageId === 'json' || document.languageId === 'jsonc')) {
+			// TODO: how to get the text editor options tabSize and insertSpaces
```

Comment:
Good question. A not so pretty but pragmatic way would be to check if the document is open in any text editor and read it from there.